packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -287,6 +300,38 @@ export const withBlockBindingSupport = createHigherOrderComponent(
 			]
 		);
 
+		// Throw a warning if the block is connected to an invalid source.
+		if ( invalidBinding ) {
+			const removeAllBindingsButton = (
+				<Button
+					__next40pxDefaultSize={ false }
+					key="remove-all-bindings"
+					onClick={ () =>
+						updateBlockBindings( {
+							[ invalidBinding.attribute ]: undefined,
+						} )
+					}
+					variant="primary"
+				>
+					{ __( 'Remove connection' ) }
+				</Button>
+			);
+			return (
+				<div className="has-warning">
+					<Warning actions={ [ removeAllBindingsButton ] }>
+						{ sprintf(
+							/* translators: %1$s: block attribute, %2$s: invalid block bindings source. */
+							__(
+								'Attribute "%1$s" is connected to undefined "%2$s" block bindings source. You can leave this block intact, modify the connection, or remove it.'
```

Comment:
Any suggestion to improve the warning message? I tried to make it similar to other warnings, but maybe it can be simplified.

Reply:
Maybe removing "block bindings" or "block" on "block bindings source"?

Reply:
I was worried about not being clear enough, but _'Attribute "content" connected to unrecognized/unregistered 
"plugin/undefined" source.'_ looks fine to me. It's true we are not mentioning bindings anywhere. I was also expecting the message to be shown to technical users.

Reply:
I like using the word *unrecognized* to explain this.

Reply:
For now, I've changed the message to use "unrecognized" and removed the mention to "block bindings". We can keep the rest of the discussion in the other thread.